gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ObservationMapper.java

```diff
@@ -93,45 +98,75 @@ public List<Observation> mapResources(RCMRMT030101UKEhrExtract ehrExtract, Patie
     }
 
     private Observation mapObservation(RCMRMT030101UKEhrComposition ehrComposition,
-                                       RCMRMT030101UKObservationStatement observationStatement, Patient patient, List<Encounter> encounters,
+                                       RCMRMT030101UKObservationStatement observationStatement,
+                                       Patient patient,
+                                       List<Encounter> encounters,
                                        String practiseCode) {
 
+        var compoundStatement = ehrComposition
+            .getComponent()
+            .stream()
+            .flatMap(component4 -> extractAllCompoundStatements(component4))
+            .filter(Objects::nonNull)
+            .findFirst().orElseGet(RCMRMT030101UKCompoundStatement::new);
```

Comment:
Just checking this is the expected behaviour, if an empty Optional is returned, we'd like to create a new instance of `RCMRMT030101UKCompoundStatement`?

Reply:
in some scenarios there might not be any compoundStatement(s) hence I thought it would be more intutitive and less code to create a temporary new CompoundStatement instance which will be instantiated with null confidentialityCode. A compoundStatement instance is only used to get confidentialityCode.